pkg/util/ciutil/detect.go

```diff
@@ -96,9 +96,11 @@ var detectors = map[SystemName]system{
 		Name:            Hudson,
 		EnvVarsToDetect: []string{"HUDSON_URL"},
 	},
-	Jenkins: baseCI{
-		Name:            Jenkins,
-		EnvVarsToDetect: []string{"JENKINS_URL", "BUILD_ID"},
+	Jenkins: jenkinsCI{
+		baseCI: baseCI{
+			Name:            Jenkins,
+			EnvVarsToDetect: []string{"JENKINS_URL"},
```

Comment:
I removed `BUILD_ID` here because it isn't a prefixed env var specific to Jenkins, which could cause us to accidentally detect it as Jenkins if some other CI system were to use that same env var.

Reply:
That seems like a reasonable change. But alternatively, could we just have `EnvVarsToDetect` only match the CI system if _all_ of them are found? Probably won't work for every system, but maybe worth considering. (Though I assume that for the 99% case, finding one env var, e.g. `JENKINS_URL` is sufficient.)

Reply:
> But alternatively, could we just have EnvVarsToDetect only match the CI system if all of them are found?

Ha. I actually thought about making that change, but as you say in the next sentence, all of the systems we currently detect have some dedicated variable that will always be set.